README.rst

```diff
@@ -0,0 +1,227 @@
+  inputs to the task and the actual definition of the task will be used to
+  determine the resultant hash, so that if, say, a script changes, the
+  results will be recomputed. This is hard to achieve with `stepIO`.
+- External tasks are naturally distributable. When running in a production setting,
+  you're likely to want to distribute tasks among multiple machines. This is
+  not, in general, achievable with arbitrary `IO` computations.
+
+FunFlow's current approach to external tasks is heavily based on Docker_. Using
+Docker allows tasks to be self-contained, and adds minimal requirements to the
+system being used to host funflow instances (they just need to have docker running!)
+
+To use a docker container as an external step, one simply needs to define a function
+of type `a -> Docker.Config`, where `a` is the input type to the flow. At its core,
+this means:
+
+- Specifying the docker image (and optionally, image ID) to use. For example,
+  `nixos/nix:1.11.14`.
+- Specifying the path to the command which will be run within the container.
+- Specifying which inputs (from the content store) are to be mounted within
+  the container.
+
+Here's an example::
+
+  myDockerStep :: SimpleFlow CS.Item CS.Item
+  myDockerStep = docker $ \input -> Docker.Config
+    { Docker.image = "nixos/nix"
+    , Docker.optImageID = Just "1.11.14"
+    , Docker.input = Docker.SingleInput input
+    , Docker.command = "myScript.sh"
+    , Docker.args =
+        [
+        , "--input_dir", "/input/"
+        , "--output_dir", "/output/"
+        ]
+    }
+
+A `CS.Item` refers to an item within the content store. You can use
+`putInStore`, `getFromStore`, `copyFileToStore` and similar tools to add and
+fetch files from the store. This lets you interleave internal and external
+computations.
+
+Running on multiple machines
+----------------------------
+
+To run on multiple machines, you need to use one of the distributable
+coordinators - either the Redis coordinator or the SQLite coordinator. To do
+this, you need to:
+
+- Start some *executors* pointed at the coordinator.
+- Run your flow using that coordinator.
+
+The simplest way to run an executor is to use the bundled `ffexecutord` executable.
+This can work with either the Redis or SQLite coordinators. Here's how we might
+initialise an executor using `/tmp/funflow` as the content store directory, and
+`/tmp/coordinator.db` as our coordinating database::
+
+  ffexecutord sqlite /tmp/funflow /tmp/coordinator.db
+
+We then need to run the flow, pointing at this coordinator. To do so, we'll need a
+slightly more complex function from Control.FunFlow.Exec.Simple_: `runSimpleFlow`. We
+need to give this the correct parameters for the `SQLite` coordinator::
+
+  CS.withStore [absdir|/tmp/funflow|] $ \store -> do
+    runFlow SQLite [absfile|/tmp/coordinator.db|] store runNoEffect 123123 flow input
+
+A couple of the parameters here may be confusing:
+
+runNoEffect
```

Comment:
I like the formatting you've used here for explaining the code snippet - could do the same with the first code snippet too, as you have a similar explanation after that.